core/src/test/java/io/grpc/MetadataTest.java

```diff
@@ -242,15 +250,15 @@ public void verifyToString() {
     Metadata h = new Metadata();
     h.put(KEY, new Fish("binary"));
     h.put(Metadata.Key.of("test", Metadata.ASCII_STRING_MARSHALLER), "ascii");
-    assertEquals("Metadata({test-bin=[Fish(binary)], test=[ascii]})", h.toString());
+    assertEquals("Metadata(test-bin=Fish(binary),test=ascii)", h.toString());
 
     Metadata t = new Metadata();
     t.put(Metadata.Key.of("test", Metadata.ASCII_STRING_MARSHALLER), "ascii");
-    assertEquals("Metadata({test=[ascii]})", t.toString());
+    assertEquals("Metadata(test=ascii)", t.toString());
 
     t = new Metadata("test".getBytes(US_ASCII), "ascii".getBytes(US_ASCII),
         "test-bin".getBytes(US_ASCII), "binary".getBytes(US_ASCII));
-    assertEquals("Metadata({test=[ascii], test-bin=[[98, 105, 110, 97, 114, 121]]})", t.toString());
+    assertEquals("Metadata(test=ascii,test-bin=binary)", t.toString());
```

Comment:
I personally think checking to toString implementation is kinda stringent, but am not ready to remove it.  This is kind of a hack because I don't think we ensure that all bytes are actually ascii at the moment (like control characters) so this may not work out.